From 199f8455745654d3856251a949ad630c4d1a3c13 Mon Sep 17 00:00:00 2001
From: Ludovic Rousseau <ludovic.rousseau@free.fr>
Date: Thu, 2 Jul 2020 19:44:36 +0200
Subject: [PATCH] Fix compiler warning
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

hotplug_libusb.c: In function ‘HPSearchHotPluggables’:
hotplug_libusb.c:498:4: warning: cast between incompatible function types from ‘void (*)(int *)’ to ‘void * (*)(void *)’ [-Wcast-function-type]
  498 |    (PCSCLITE_THREAD_FUNCTION( )) HPEstablishUSBNotifications, pipefd);
      |    ^
---
 src/hotplug_libusb.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/hotplug_libusb.c b/src/hotplug_libusb.c
index 996bc0c..22762a2 100644
--- a/src/hotplug_libusb.c
+++ b/src/hotplug_libusb.c
@@ -403,7 +403,7 @@ static void HPRescanUsbBus(void)
 	libusb_free_device_list(devs, 1);
 }
 
-static void HPEstablishUSBNotifications(int pipefd[2])
+static void * HPEstablishUSBNotifications(int pipefd[2])
 {
 	int i, do_polling;
 	int r;
@@ -415,7 +415,7 @@ static void HPEstablishUSBNotifications(int pipefd[2])
 		Log2(PCSC_LOG_CRITICAL, "libusb_init failed: %d", r);
 		/* emergency exit */
 		kill(getpid(), SIGTERM);
-		return;
+		return NULL;
 	}
 
 	/* scan the USB bus for devices at startup */
@@ -470,6 +470,8 @@ static void HPEstablishUSBNotifications(int pipefd[2])
 		close(rescan_pipe[0]);
 		rescan_pipe[0] = -1;
 	}
+
+	return NULL;
 }
 
 LONG HPSearchHotPluggables(void)
-- 
2.30.0.284.gd98b1dd5eaa7-goog

